scipy/optimize/_linprog.py

```diff
@@ -403,17 +402,13 @@ def linprog(c, A_ub=None, b_ub=None, A_eq=None, b_eq=None,
     `'highs-ds'` and
     `'highs-ipm'` are interfaces to the
     HiGHS simplex and interior-point method solvers [13]_, respectively.
-    `'highs'` chooses between
+    `'highs'` (default) chooses between
     the two automatically. These are the fastest linear
     programming solvers in SciPy, especially for large, sparse problems;
     which of these two is faster is problem-dependent.
-    `'interior-point'` is the default
-    as it was the fastest and most robust method before the recent
-    addition of the HiGHS solvers.
-    `'revised simplex'` is more
-    accurate than interior-point for the problems it solves.
-    `'simplex'` is the legacy method and is
-    included for backwards compatibility and educational purposes.
+    The other solvers (`'interior-point'`, `'revised simplex'`, and
+    `'simplex'`) are legacy methods and will be removed in the second release
+    after SciPy 1.9.0.
```

Comment:
"second release after SciPy 1.9.0." is not clear - presumably this doesn't refer to 1.9.2, but 1.11.0? Unless there are concrete plans for a scipy 2.0, I'd just write 1.11.0 here, or at least qualify "second non-patch release after 1.9.0".

Reply:
Thanks... yeah, it would have been "1.11.0" but for comments in gh-13490. I was thinking about qualifying with "minor" but was worried some users would think that would mean bug-fix release. Is there a standard description other than "non-patch"? "feature" release?

Reply:
OK, I looked at that [discussion](https://github.com/scipy/scipy/pull/13490#discussion_r790426384), but I kinda fail to see the reason why the minor counter couldn't go to two digits - numpy and cpython do the same...? I'm thinking a putative 2.0 should come with some degree of "large change compared to `1.x`", even if it isn't breaking anything**.

In any case, I'd say it would be simpler to just leave 1.11 here, and if a version 2.0 is decided (should we open an issue to discuss?), then it's a simple matter of searching for "1.10"/"1.11"/etc. and replacing them appropriately.

CC @ev-br @rgommers @tupui @tylerjereddy 

> Is there a standard description other than "non-patch"? "feature" release?

I wanted to avoid "minor" due to not mapping well to scipy's versioning scheme, but "feature release" would work. That's still worse than using 1.11 IMO, but definitely an improvement over what we have now.

** as an example in terms of the kind of scope for a major bump that I have in mind, a current(ly-talked-about) candidate for cpython 4.0 is if/once the GIL is removed. On the numpy side, although there is an ancient, non-empty [milestone](https://github.com/numpy/numpy/milestone/9), I'm not aware of current plans for 2.0 (though I did once [wager](https://github.com/numpy/numpy/issues/12525#issuecomment-452463450) that it'll come about eventually 😉) - comprehensively fixing integer promotion issues _might_ be a candidate to go to numpy 2.0.

Reply:
I'd say "two releases after N" and leave it at that

Reply:
> I'd say "two releases after N" and leave it at that

That's the part I disagree with. Strictly speaking "two releases after 1.9.0" is "1.9.2", which is clearly not what is intended here.

Reply:
If you feel strongly enough to conflate the issue at hand with a decision on the 1.10 vs 2.0 naming, I suggest to separate the latter and discuss it on the mailing list. 

My personal opinion is that 2.0 is no special and I'd have a slight preference for 1.9->2.0. That said, it's not quite my hill so I'd go with a majority on this one :-)

Reply:
Respectfully, I'm not conflating anything. Avoiding a direct "will be removed in Scipy 1.11.0" and ending up with awkward wording (i.e. the issue at hand) _only_ makes sense if a possible 2.0 release is in play. Still, I'll open an issue to discuss the version after 1.9.

Reply:
> Still, I'll open an issue to discuss the version after 1.9.

#15528